libs/blockscout-auth/src/auth.rs

```diff
@@ -0,0 +1,347 @@
+use cookie::Cookie;
+use reqwest::{
+    header::{HeaderMap, HeaderValue},
+    StatusCode,
+};
+use serde::Deserialize;
+use std::collections::HashMap;
+use thiserror::Error;
+use tonic::{codegen::http::header::COOKIE, metadata::MetadataMap};
+use url::Url;
+
+const JWT_TOKEN_NAME: &str = "_explorer_key";
+const CSRF_TOKEN_NAME: &str = "x-csrf-token";
+const API_KEY_NAME: &str = "api_key";
+
+#[derive(Debug)]
+pub struct AuthSuccess {
+    pub user_id: String,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(untagged)]
+enum BlockscoutAuthResponse {
+    Success(BlockscoutAuthSuccessResponse),
+    Error(BlockscoutAuthErrorResponse),
+}
+
+#[derive(Debug, Clone, Deserialize)]
+struct BlockscoutAuthErrorResponse {
+    message: String,
+}
```

Comment:
Wouldn't the request result in non 200 status code in that case?

Reply:
If blockscout returned 200 with {"message": ""} then yes, otherwise no

Reply:
And what is the case when 200 with {"message": ""} is returned?